src/jshint.js

```diff
@@ -3668,6 +3682,9 @@ var JSHINT = (function() {
         continue;
       }
 
+      var superCall = c.heritage && c.heritage.type !== "null" &&
```

Comment:
it doesn't catch this highly contrived example
```
// jshint esnext: true
class a extends function(){ return null;}() {
  constructor() {
    super();
  }
}
```
but I think that is fine. If we take that into account we can't really do any static analysis of whether super is required/allowed.